docs/new-provider.md

```diff
@@ -0,0 +1,53 @@
+# Developing a New Provider
+
+It is relatively easy to adapt a Terraform Provider, X, for use with Pulumi.  The
+[Cloudflare provider](https://github.com/pulumi/pulumi-cloudflare) offers a good blueprint for how to go about this.
```

Comment:
Should this replace https://github.com/pulumi/pulumi-terraform-bridge/tree/master/pf/README.md perhaps ?

Reply:
I'll add a section pointing to pf/README.md for bridging PF based providers. We can take moving it in another PR.